src/mame/drivers/vii.cpp

```diff
@@ -609,90 +635,123 @@ void vii_state::switch_bank(uint32_t bank)
 	}
 }
 
-void vii_state::do_gpio(uint32_t offset)
+uint16_t spg2xx_cart_state::do_spg243_vii_io(uint16_t what, int index)
 {
-	uint32_t index  = (offset - 1) / 5;
-	uint16_t buffer = m_io_regs[5*index + 2];
-	uint16_t dir    = m_io_regs[5*index + 3];
-	uint16_t attr   = m_io_regs[5*index + 4];
-	uint16_t special= m_io_regs[5*index + 5];
+	if(index == 1)
+	{
+		uint32_t bank = ((what & 0x80) >> 7) | ((what & 0x20) >> 4);
+		switch_bank(bank);
+	}
+	return what;
+}
 
-	uint16_t push   = dir;
-	uint16_t pull   = (~dir) & (~attr);
-	uint16_t what   = (buffer & (push | pull));
-	what ^= (dir & ~attr);
-	what &= ~special;
+uint16_t spg2xx_cart_state::do_spg243_vsmile_io(uint16_t what, int index)
+{
+	// TODO: find out how vsmile accesses these GPIO regs!
+	return what;
+}
 
-	if (m_spg243_mode == SPG243_VII)
+
+
+uint16_t spg2xx_game_state::do_spg243_batman_io(uint16_t what, int index)
+{
+	if (index == 0)
 	{
-		if(index == 1)
-		{
-			uint32_t bank = ((what & 0x80) >> 7) | ((what & 0x20) >> 4);
-			switch_bank(bank);
-		}
+		uint16_t temp = m_io_p1->read();
+		what |= (temp & 0x0001) ? 0x8000 : 0;
+		what |= (temp & 0x0002) ? 0x4000 : 0;
+		what |= (temp & 0x0004) ? 0x2000 : 0;
+		what |= (temp & 0x0008) ? 0x1000 : 0;
+		what |= (temp & 0x0010) ? 0x0800 : 0;
+		what |= (temp & 0x0020) ? 0x0400 : 0;
+		what |= (temp & 0x0040) ? 0x0200 : 0;
+		what |= (temp & 0x0080) ? 0x0100 : 0;
 	}
-	else if (m_spg243_mode == SPG243_BATMAN)
+
+	if (index == 2)
 	{
-		if(index == 0)
-		{
-			uint16_t temp = m_io_p1->read();
-			what |= (temp & 0x0001) ? 0x8000 : 0;
-			what |= (temp & 0x0002) ? 0x4000 : 0;
-			what |= (temp & 0x0004) ? 0x2000 : 0;
-			what |= (temp & 0x0008) ? 0x1000 : 0;
-			what |= (temp & 0x0010) ? 0x0800 : 0;
-			what |= (temp & 0x0020) ? 0x0400 : 0;
-			what |= (temp & 0x0040) ? 0x0200 : 0;
-			what |= (temp & 0x0080) ? 0x0100 : 0;
-		}
+	}
+	return what;
+}
 
-		if(index == 2)
-		{
-		}
+uint16_t spg2xx_game_state::do_spg240_rad_skat_io(uint16_t what, int index)
+{
+	// have not checked for outputs yet.
+
+	if (index == 0)
+	{
+		what = m_io_p1->read();
 	}
-	else if (m_spg243_mode == SPG243_VSMILE)
+	else if (index == 1)
 	{
-		// TODO: find out how vsmile accesses these GPIO regs!
+		what = m_io_p2->read();
 	}
-	else if (m_spg243_mode == SPG243_WIRELESS60)
+	else if (index == 2)
 	{
-		if(index == 0)
-		{
-			switch(what & 0x300)
-			{
-				case 0x300:
-					m_w60_controller_input = -1;
-					break;
-
-				case 0x200:
-					m_w60_controller_input++;
-					break;
-
-				default:
-					uint16_t temp1 = m_io_p1->read();
-					uint16_t temp2 = m_io_p2->read();
-					uint16_t temp3 = 1 << m_w60_controller_input;
-					if (temp1 & temp3) what ^= 0x400;
-					if (temp2 & temp3) what ^= 0x800;
-					break;
-			}
-		}
+		what = m_io_p3->read();
+	}
+	return what;
+}
 
-		if(index == 1)
+uint16_t spg2xx_game_state::do_spg243_wireless60_io(uint16_t what, int index)
+{
+	if(index == 0)
```

Comment:
Since you're touching the code anyway, could you please make the spacing on the `if` and `switch` here consistent?